File: backend/src/main/scala/microservices/feedback/objects/SendFeedbackMessagePlannerRequest.scala
// SendFeedbackMessagePlannerRequest：feedback 域后端对象定义。
package com.typesafe.travel.feedback.domain
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class SendFeedbackMessagePlannerRequest(threadId: String, senderDisplayName: String, senderRole: String, body: String)
object SendFeedbackMessagePlannerRequest:
  given sourceEncoder: Encoder[SendFeedbackMessagePlannerRequest] = deriveEncoder
  given sourceDecoder: Decoder[SendFeedbackMessagePlannerRequest] = deriveDecoder

File: frontend/src/microservices/feedback/objects/SendFeedbackMessagePlannerRequest.ts
// 本文件定义 feedback 模块的 `SendFeedbackMessagePlannerRequest`，用于提交一条普通文本反馈消息。

import type { FeedbackSenderRole } from './FeedbackSenderRole'

export type SendFeedbackMessagePlannerRequest = {
  senderRole: FeedbackSenderRole
  senderDisplayName: string
  body: string
}
export const sendFeedbackMessageRequestFromJson = (json: string): SendFeedbackMessagePlannerRequest =>
  JSON.parse(json) as SendFeedbackMessagePlannerRequest

export const sendFeedbackMessageRequestToJson = (value: SendFeedbackMessagePlannerRequest): string =>
  JSON.stringify(value)